CMakeLists.txt

```diff
@@ -96,39 +97,43 @@ set_option_if(ACTS_BUILD_PLUGIN_AUTODIFF ACTS_BUILD_EVERYTHING)
 include(GNUInstallDirs) # GNU-like installation paths, e.g. lib/, include/, ...
 include(ActsCompilerOptions) # default compile options
 include(ActsComponentsHelpers) # handle components via add_..._if commands
-include(TargetSourcesLocal) # needed only until we require CMake 3.13
 # place build products in `<build_dir>/bin` and `<build_dir>/lib` for simple use
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
 
+# dependency versions
+set(_acts_autodiff_version 0.5.11)
+set(_acts_boost_version 1.71)
+set(_acts_dd4hep_version 1.11)
+set(_acts_doxygen_version 1.8.15)
+set(_acts_eigen3_version 3.3.7)
+set(_acts_hepmc3_version 3.2.1)
+set(_acts_nlohmanjson_version 3.2.0)
+set(_acts_root_version 6.20)
+set(_acts_tbb_version 2020.1)
+
 # required packages
-# use only the CMake FindBoost module; avoids issues with the new cmake configs
-# that become available with Boost 1.70
-set(Boost_NO_BOOST_CMAKE ON)
-find_package(Boost 1.69 MODULE REQUIRED COMPONENTS filesystem program_options unit_test_framework)
-find_package(Eigen3 3.2.9 CONFIG REQUIRED)
 # NOTE we use boost::filesystem instead of std::filesystem since the later
 #   is not uniformly supported even on compilers that nominally support C++17 
-
-# Helper macro that overrides the project function in subdirectories, so 
-# we can intercept and store their versions. This is used to lock them down
-# in an installation later on.
-cmake_policy(SET CMP0048 NEW)    
-macro(project)
-    _project(${ARGN})
-    set(${PROJECT_NAME}_VERSION "${${PROJECT_NAME}_VERSION}" CACHE INTERNAL "")
-endmacro()
```

Comment:
Is this not needed anymore?

Reply:
Yes, from what I have seen. The CMake policy is related to `PROJECT_VERSION` which we are not using anywhere; only the `<project_name>_VERSION` is used by us and is not affected.

The macro to reset `<project_name>_VERSION` seems to be unnecessary as well. The relevant variables set by the regular `project(...)` calls are globally available and their values are used to fill e.g. the `ActsConfig.cmake` file (at least on my machine).

Reply:
No the problem was that when we `add_subdirectory` autodiff, the variable isn't set / available. And if we want to lock down the version in our installed `Config.cmake` we need to bake it in. That's why I added this.

Reply:
The something else must have change, because with the changes the generated `ActsConfig.cmake` file contains the correct `autodiff_VERSION` variable.

Reply:
See #592. I think this is **only** if you build the bundled version.

Reply:
Nevermind. After starting with a clean CMake cache, I see exactly the issue that you describe. I will remove the removal.

Reply:
I readded the redefinition and extended the docs.